### PR TITLE
CSPL-2299: Ginkgo v2

### PR DESCRIPTION
**Pull request by vivekr-splunk**
moving to ginkgo v2

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -126,7 +126,9 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test  -v -covermode=count -coverprofile=coverage.out --timeout=300s   ./pkg/splunk/common ./pkg/splunk/enterprise ./pkg/splunk/controller ./pkg/splunk/client ./pkg/splunk/util ./controllers 
+	#KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test  -v -covermode=count -coverprofile=coverage.out --timeout=300s   ./pkg/splunk/common ./pkg/splunk/enterprise ./pkg/splunk/controller ./pkg/splunk/client ./pkg/splunk/util ./controllers  2>&1 | tee coverage.out
```
Comment (gaurav-splunk) [inline Makefile:129]: remove this line?

File: controllers/suite_test.go [hunk 2]
```diff
@@ -17,18 +17,20 @@ limitations under the License.
...
-	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"go.uber.org/zap/zapcore"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
-	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	//"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
```
Comment (gaurav-splunk) [inline controllers/suite_test.go:33]: remove this line?

File: test/run-tests.sh [hunk 3]
```diff
@@ -191,4 +191,5 @@ fi
 echo "Skipping following test :: ${TEST_TO_SKIP}"
 
 # Running only smoke test cases by default or value passed through TEST_FOCUS env variable. To run different test packages add/remove path from focus argument or TEST_FOCUS variable
-ginkgo -v -keepGoing --trace -progress -r -nodes=${CLUSTER_NODES} --noisyPendings=false --reportPassed --focus="${TEST_TO_RUN}" --skip="${TEST_TO_SKIP}" ${topdir}/test/ -- -commit-hash=${COMMIT_HASH} -operator-image=${PRIVATE_SPLUNK_OPERATOR_IMAGE}  -splunk-image=${PRIVATE_SPLUNK_ENTERPRISE_IMAGE} -cluster-wide=${CLUSTER_WIDE}
\ No newline at end of file
+echo "ginkgo --junit-report=inttest.xml -vv --keep-going --trace -r --timeout=3h  -nodes=${CLUSTER_NODES} --focus="${TEST_TO_RUN}" --skip="${TEST_TO_SKIP}" --output-interceptor-mode=none --cover ${topdir}/test/ -- -commit-hash=${COMMIT_HASH} -operator-image=${PRIVATE_SPLUNK_OPERATOR_IMAGE}  -splunk-image=${PRIVATE_SPLUNK_ENTERPRISE_IMAGE} -cluster-wide=${CLUSTER_WIDE}"
```
Comment (gaurav-splunk) [inline test/run-tests.sh:194]: did we add this for debugging purpose and should it be removed?

Comment (vivekr-splunk) [reply test/run-tests.sh:194]: we added for debugging

File: controllers/suite_test.go [hunk 4]
```diff
@@ -48,12 +49,11 @@ var k8sManager ctrl.Manager
 func TestAPIs(t *testing.T) {
 	RegisterFailHandler(Fail)
 
-	RunSpecsWithDefaultAndCustomReporters(t,
```
Comment (akondur) [inline controllers/suite_test.go:51]: Has this API been deprecated? Or changed to include the reporter(NewlineReporter) by default?

File: test/appframework_aws/c3/appframework_aws_suite_test.go [hunk 5]
```diff
@@ -56,8 +55,7 @@ func TestBasic(t *testing.T) {
 
 	RegisterFailHandler(Fail)
 
-	junitReporter := reporters.NewJUnitReporter(testSuiteName + "_junit.xml")
-	RunSpecsWithDefaultAndCustomReporters(t, "Running "+testSuiteName, []Reporter{junitReporter})
+	RunSpecs(t, "Running "+testSuiteName)
```
Comment (akondur) [inline test/appframework_aws/c3/appframework_aws_suite_test.go:58]: Same comment as above

**Review by akondur — APPROVED** [review]
Looks good. Left minor comment.

File: controllers/suite_test.go (hunk 4 above)
Comment (vivekr-splunk) [reply controllers/suite_test.go:51]: yes its depricated, by default it provides junit report

**Pull request merged**